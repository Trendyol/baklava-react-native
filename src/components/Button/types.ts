export type ButtonVariantVariantTypes = 'primary' | 'secondary' | 'tertiary';

export type ButtonKindVariantTypes =
  | 'default'
  | 'neutral'
  | 'success'
  | 'danger';

export type ButtonSizeVariantTypes = 's' | 'm' | 'l';
