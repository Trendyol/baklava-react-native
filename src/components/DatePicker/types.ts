export type Day = {
  day: number;
  month: number;
  year: number;
  isActive: boolean;
  isPrevMonth: boolean;
  isNextMonth: boolean;
  isCurrentDate: boolean;
  timestamp: number;
  yearWithMonth: number;
};

export interface CalendarData {
  calendar: (string[] | Day[])[];
  year: number;
  month: number;
  nameOfMonth: string;
}

export type CalendarOptions = {
  year?: number;
  month?: number;
  nameOfWeekdays?: string[];
  nameOfMonths?: string[];
  firstDayOfWeek?: 0 | 1;
};
