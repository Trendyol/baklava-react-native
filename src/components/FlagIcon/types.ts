import { flagIconList } from './list';

export type FlagIconNameType = (typeof flagIconList)[number];
