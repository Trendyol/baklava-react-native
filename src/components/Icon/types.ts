import { iconList } from './list';

export type IconNameType = typeof iconList[number];
