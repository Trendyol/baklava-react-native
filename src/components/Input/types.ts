import { EasingFunction, TextInput } from 'react-native';

export type TextInputRefType = TextInput;

export type InputRefType =
  () => React.MutableRefObject<TextInputRefType | null>;

export type AnimatedViewPropsType = {};

export type AnimatedTextPropsType = {
  style?: any;
};

export type CommonAnimatedPropsTypes = {
  duration: number;
  useNativeDriver: boolean;
  easing: EasingFunction;
};
