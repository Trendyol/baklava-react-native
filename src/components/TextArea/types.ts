import { EasingFunction, TextInput } from 'react-native';

export type TextInputRefType = TextInput;

export type InputRefType =
  () => React.MutableRefObject<TextInputRefType | null>;

export type AnimatedViewPropsType = {};

export type AnimatedTextPropsType = {};

export type CommonAnimatedPropsTypes = {
  duration: number;
  useNativeDriver: boolean;
  easing: EasingFunction;
};
