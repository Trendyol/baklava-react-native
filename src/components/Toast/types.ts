import { VariantProps } from '@ergenekonyigit/restyle';
import { Theme } from '../../theme';

export interface BaseToastData {
  /**
   * variant
   * what color will appear on the screen
   * @type {ToastVariant}
   * @default  ToastVariant.DEFAULT
   * @success ToastVariant.SUCCESS
   * @error - ToastVariant.ERROR
   */
  variant?: VariantProps<Theme, 'toastVariants'>['variant'];
  /**
   * message to appear in toast
   *
   * max length is 110
   */
  text?: string;
  /**
   * how long(ms) will it be visible on the screen
   * @default 7000
   */
  duration?: number;
  /**
   * offset from the bottom of the screen (in px)
   *
   */
  bottomOffset?: number;
}

export interface ToastDataWithoutAction extends BaseToastData {
  action?: never;
  actionText?: never;
}

export interface ToastDataWithAction extends BaseToastData {
  action: (close: () => void) => void;
  actionText: string;
}

export type ToastData = ToastDataWithoutAction | ToastDataWithAction;
export type ToastDataWithId = (ToastDataWithoutAction | ToastDataWithAction) & {
  id: string;
};
export type ShowMethodParams = ToastDataWithoutAction | ToastDataWithAction;

export const TostDurations = {
  SHORT: 7000,
  LONG: 10000,
} as const;

export const ToastDurations = {
  SHORT: 7000,
  LONG: 10000,
} as const;

export type DurationTypes = typeof ToastDurations;
export type Duration = DurationTypes[keyof DurationTypes];

export const ToastVariants = {
  DEFAULT: 'default',
  SUCCESS: 'success',
  ERROR: 'error',
  WARNING: 'warning',
} as const;

export type ToastVariantsTypes = typeof ToastVariants;
export type ToastVariant = ToastVariantsTypes[keyof ToastVariantsTypes];
