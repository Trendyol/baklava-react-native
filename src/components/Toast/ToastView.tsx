import React from 'react';
import {
  Animated,
  LayoutChangeEvent,
  LayoutRectangle,
  PanResponder,
  TouchableOpacity,
  useWindowDimensions,
} from 'react-native';
import Text from '../Text/Text';
import { TostDurations } from './types';
import { capitalizeFirstLetter, getFirstNChar, pausableTimer } from './utils';
import Box from '../Box/Box';
import {
  createRestyleComponent,
  createVariant,
  VariantProps,
} from '@ergenekonyigit/restyle';
import { Theme } from '../../theme';

type ToastViewProps = React.ComponentProps<typeof Animated.View> & {
  variant?: VariantProps<Theme, 'toastVariants'>['variant'];
  text?: string;
  onDisappared?: () => void;
  duration?: number;
  action?: (close: () => void) => void;
  actionText?: string;
};

const variantVariant = createVariant<Theme, 'toastVariants', 'variant'>({
  property: 'variant',
  themeKey: 'toastVariants',
});

const ToastViewContainer = createRestyleComponent<
  ToastViewProps & VariantProps<Theme, 'toastVariants'>,
  Theme
>([variantVariant], Animated.View);

export type ToastViewRef = {
  show: () => void;
  hide: () => void;
};

const ToastView = React.forwardRef<ToastViewRef, ToastViewProps>(
  (
    {
      variant = 'default',
      text,
      duration = TostDurations.SHORT,
      onDisappared,
      action,
      actionText,
      ...rest
    },
    ref,
  ) => {
    const dims = useWindowDimensions();
    const anim = React.useRef(new Animated.Value(dims.height / 2)).current;
    const layoutSizesRef = React.useRef<LayoutRectangle>();
    const timerRef = React.useRef(
      pausableTimer(() => {
        playHideAnim();
      }, duration),
    );

    const panResponderRef = React.useRef(
      PanResponder.create({
        onStartShouldSetPanResponderCapture: (e, gestureState) => {
          return !action || !(gestureState.dx === 0 && gestureState.dy === 0);
        },
        onMoveShouldSetPanResponder: () => true,
        onPanResponderGrant: () => {
          timerRef.current.pause();
        },
        onPanResponderMove: (e, gestureState) => {
          if (gestureState.dy >= 0) {
            Animated.event([null, { dy: anim }], { useNativeDriver: false })(
              e,
              gestureState,
            );
          }
        },
        onPanResponderRelease: (e, gestureState) => {
          if (
            gestureState.dy > (layoutSizesRef.current?.height ?? 0) ||
            gestureState.vy > 0.05
          ) {
            playHideAnim();
          } else {
            playShowAnim();
            timerRef.current.resume(2000);
          }
        },
      }),
    );

    React.useImperativeHandle(ref, () => ({
      show() {
        playShowAnim();
      },
      hide() {
        playHideAnim();
      },
    }));

    const playShowAnim = React.useCallback(() => {
      Animated.timing(anim, {
        toValue: 0,
        duration: 333,
        useNativeDriver: true,
      }).start();
    }, [anim]);

    const playHideAnim = React.useCallback(() => {
      Animated.timing(anim, {
        toValue: dims.height / 2,
        duration: 333,
        useNativeDriver: true,
      }).start(onDisappared);
    }, [anim, dims.height, onDisappared]);

    React.useEffect(() => {
      playShowAnim();
      return () => {
        timerRef!.current.kill();
      };
    }, [playShowAnim, timerRef]);

    const onLayout = React.useCallback((event: LayoutChangeEvent) => {
      layoutSizesRef.current = event.nativeEvent.layout;
    }, []);

    const actionView = React.useMemo(() => {
      if (!action) {
        return null;
      }
      return (
        <TouchableOpacity
          onPress={() => action(playHideAnim)}
          testID="toast-action"
          accessibilityLabel="toast-action">
          <Text
            variant="subtitle03Medium"
            color="contentPrimary"
            testID="toast-action-text"
            accessibilityLabel="toast-action-text"
            p="2xs">
            {capitalizeFirstLetter(actionText)}
          </Text>
        </TouchableOpacity>
      );
    }, [action, actionText, playHideAnim]);

    return (
      <ToastViewContainer
        {...panResponderRef.current.panHandlers}
        variant={variant}
        onLayout={onLayout}
        testID="toast-container"
        accessibilityLabel="toast-container"
        style={{ transform: [{ translateY: anim }] }}
        {...rest}>
        <Box flex={1} shadowRadius={30}>
          <Text
            variant="bodyText"
            color="contentPrimary"
            testID="toast-text"
            accessibilityLabel="toast-text"
            textAlign="left">
            {getFirstNChar(text!, 110, '...')}
          </Text>
        </Box>
        {actionView}
      </ToastViewContainer>
    );
  },
);

export default ToastView;
