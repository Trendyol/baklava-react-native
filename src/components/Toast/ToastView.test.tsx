// @ts-ignore
import { toHaveStyle } from '@testing-library/jest-native';
import React from 'react';
import { cleanup, fireEvent, mockPanResponder, render } from '../../test-utils';
import ToastView, { ToastViewRef } from './ToastView';
import { ToastData } from './types';

describe('ToastView', () => {
  expect.extend({ toHaveStyle });

  beforeEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  test('should render toast correctly', () => {
    // given
    const data: ToastData = {
      text: 'lorem ipsum dolor sit amet.',
    };

    // when
    const { toJSON } = render(
      <ToastView variant={data?.variant} text={data?.text} />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render toast with action correctly', () => {
    // given
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: () => null,
      actionText: 'Tamam',
    };

    // when
    const { toJSON } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        action={data.action}
        actionText={data.actionText}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render given text correctly', () => {
    // given
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
    };

    // when
    const { getByText, queryByTestId } = render(
      <ToastView variant={data?.variant} text={data?.text} />,
    );
    const text = getByText('lorem ipsum dolor sit amet.');
    const actionBtn = queryByTestId('toast-action');

    // then
    expect(actionBtn).toBeFalsy();
    expect(text).toBeTruthy();
  });

  test('should render given text and action text correctly', () => {
    // given
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByText, queryByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        action={data.action}
        actionText={data.actionText}
      />,
    );
    const text = getByText('lorem ipsum dolor sit amet.');
    const actionText = getByText('Tamam');
    const actionBtn = queryByTestId('toast-action');

    // then
    expect(actionBtn).toBeTruthy();
    expect(text).toBeTruthy();
    expect(actionText).toBeTruthy();
  });

  test('should render the default toast with color #FEF2E8', () => {
    // given
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        action={data.action}
        actionText={data.actionText}
      />,
    );
    const toastContainer = getByTestId('toast-container');
    // then
    expect(toastContainer).toHaveStyle({
      backgroundColor: '#FEF2E8',
    });
  });

  test('should render the success toast with color #E7F9EF', () => {
    // given
    const data: ToastData = {
      variant: 'success',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        action={data.action}
        actionText={data.actionText}
      />,
    );
    const toastContainer = getByTestId('toast-container');
    // then
    expect(toastContainer).toHaveStyle({
      backgroundColor: '#E7F9EF',
    });
  });

  test('should render the error toast with color #FFEEEC', () => {
    // given
    const data: ToastData = {
      variant: 'error',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        action={data.action}
        actionText={data.actionText}
      />,
    );
    const toastContainer = getByTestId('toast-container');
    // then
    expect(toastContainer).toHaveStyle({
      backgroundColor: '#FFEEEC',
    });
  });

  test('should render the warning toast with color #FFF8E6', () => {
    // given
    const data: ToastData = {
      variant: 'warning',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        action={data.action}
        actionText={data.actionText}
      />,
    );
    const toastContainer = getByTestId('toast-container');
    // then
    expect(toastContainer).toHaveStyle({
      backgroundColor: '#FFF8E6',
    });
  });

  test('should handle the action button press event', () => {
    const mockAction = jest.fn();
    // given
    const data: ToastData = {
      variant: 'error',
      text: 'lorem ipsum dolor sit amet.',
      action: mockAction,
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        action={data.action}
        actionText={data.actionText}
      />,
    );
    const toastAction = getByTestId('toast-action');
    fireEvent.press(toastAction);
    // then
    expect(mockAction).toBeCalled();
  });

  test('should handle onLayout', () => {
    const mockAction = jest.fn();
    // given
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: mockAction,
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        action={data.action}
        actionText={data.actionText}
      />,
    );
    const toastContainer = getByTestId('toast-container');
    fireEvent(toastContainer, 'layout', {
      nativeEvent: { layout: { height: 200 } },
    });
  });

  test('should play hide anim after 7 seconds', () => {
    jest.useFakeTimers();
    const mockAction = jest.fn();
    const mockOnDisappared = jest.fn();
    // given
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: mockAction,
      actionText: 'Tamam',
    };

    // when
    render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        duration={data.duration!}
        action={data.action}
        actionText={data.actionText}
        onDisappared={mockOnDisappared}
      />,
    );
    jest.runAllTimers();
    expect(mockOnDisappared).toBeCalled();
  });

  test('should handle ref show method', () => {
    const toastRef = React.createRef<ToastViewRef>();
    const mockAction = jest.fn();
    const mockOnDisappared = jest.fn();
    // given
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: mockAction,
      actionText: 'Tamam',
    };

    // when
    render(
      <ToastView
        ref={toastRef}
        variant={data?.variant}
        text={data?.text}
        duration={data.duration!}
        action={data.action}
        actionText={data.actionText}
        onDisappared={mockOnDisappared}
      />,
    );
    toastRef.current?.show();
  });

  test('should handle ref hide method', () => {
    const toastRef = React.createRef<ToastViewRef>();
    const mockAction = jest.fn();
    const mockOnDisappared = jest.fn();
    // given
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: mockAction,
      actionText: 'Tamam',
    };

    // when
    render(
      <ToastView
        ref={toastRef}
        variant={data?.variant}
        text={data?.text}
        duration={data.duration!}
        action={data.action}
        actionText={data.actionText}
        onDisappared={mockOnDisappared}
      />,
    );
    toastRef.current?.hide();
  });

  test('should handle onStartShouldSetPanResponderCapture', async () => {
    // given
    mockPanResponder();
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        action={data.action}
        actionText={data.actionText}
      />,
    );

    const panHandler = getByTestId('toast-container');
    panHandler?.props.onStartShouldSetResponderCapture({}, { dx: 0, dy: 0 });
  });

  test('should handle onResponderGrant', async () => {
    // given
    mockPanResponder();
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView variant={data?.variant} text={data?.text} />,
    );

    const panHandler = getByTestId('toast-container');
    panHandler?.props.onResponderGrant();
  });

  test('should handle onResponderMove', async () => {
    // given
    mockPanResponder();
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView variant={data?.variant} text={data?.text} />,
    );

    const panHandler = getByTestId('toast-container');
    panHandler?.props.onResponderMove({}, { dy: 10 });
  });

  test('should handle onResponderRelease swipe bigger than height', async () => {
    jest.useFakeTimers();
    // given
    const mockOnDisappared = jest.fn();
    mockPanResponder();
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        onDisappared={mockOnDisappared}
      />,
    );

    const toastContainer = getByTestId('toast-container');

    fireEvent(toastContainer, 'layout', {
      nativeEvent: { layout: { height: 100 } },
    });

    toastContainer?.props.onResponderRelease({}, { dy: 110 });
    jest.advanceTimersByTime(350);
    expect(mockOnDisappared).toBeCalled();
  });

  test('should handle onResponderRelease swipe lower than height', async () => {
    jest.useFakeTimers();
    // given
    const mockOnDisappared = jest.fn();
    mockPanResponder();
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        onDisappared={mockOnDisappared}
      />,
    );

    const toastContainer = getByTestId('toast-container');

    fireEvent(toastContainer, 'layout', {
      nativeEvent: { layout: { height: 100 } },
    });

    toastContainer?.props.onResponderRelease({}, { dy: 90 });
    jest.advanceTimersByTime(350);
    expect(mockOnDisappared).not.toBeCalled();
  });

  test('should handle onResponderRelease high velocity', async () => {
    jest.useFakeTimers();
    // given
    const mockOnDisappared = jest.fn();
    mockPanResponder();
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        onDisappared={mockOnDisappared}
      />,
    );

    const toastContainer = getByTestId('toast-container');

    fireEvent(toastContainer, 'layout', {
      nativeEvent: { layout: { height: 100 } },
    });

    toastContainer?.props.onResponderRelease({}, { dy: 10, vy: 0.051 });
    jest.advanceTimersByTime(350);
    expect(mockOnDisappared).toBeCalled();
  });

  test('should handle onResponderRelease low velocity', async () => {
    jest.useFakeTimers();
    // given
    const mockOnDisappared = jest.fn();
    mockPanResponder();
    const data: ToastData = {
      variant: 'default',
      text: 'lorem ipsum dolor sit amet.',
      action: jest.fn(),
      actionText: 'Tamam',
    };

    // when
    const { getByTestId } = render(
      <ToastView
        variant={data?.variant}
        text={data?.text}
        onDisappared={mockOnDisappared}
      />,
    );

    const toastContainer = getByTestId('toast-container');

    fireEvent(toastContainer, 'layout', {
      nativeEvent: { layout: { height: 100 } },
    });

    toastContainer?.props.onResponderRelease({}, { dy: 10, vy: 0.049 });
    jest.advanceTimersByTime(350);
    expect(mockOnDisappared).not.toBeCalled();
  });
});
