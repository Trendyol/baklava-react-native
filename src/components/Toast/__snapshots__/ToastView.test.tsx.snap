// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastView should render toast correctly 1`] = `
<View
  accessibilityLabel="toast-container"
  collapsable={false}
  onLayout={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FEF2E8",
      "borderColor": "#F27A1A",
      "borderRadius": 6,
      "borderWidth": 1,
      "elevation": 5,
      "flexDirection": "row",
      "minHeight": 40,
      "paddingHorizontal": 12,
      "paddingVertical": 8,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 10,
      "transform": [
        {
          "translateY": 667,
        },
      ],
    }
  }
  testID="toast-container"
>
  <View
    accessible={false}
    style={
      [
        {
          "flex": 1,
          "shadowRadius": 30,
        },
      ]
    }
  >
    <Text
      accessibilityLabel="toast-text"
      accessible={true}
      style={
        [
          {
            "color": "#273142",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 16,
            "textAlign": "left",
          },
        ]
      }
      testID="toast-text"
    >
      lorem ipsum dolor sit amet.
    </Text>
  </View>
</View>
`;

exports[`ToastView should render toast with action correctly 1`] = `
<View
  accessibilityLabel="toast-container"
  collapsable={false}
  onLayout={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FEF2E8",
      "borderColor": "#F27A1A",
      "borderRadius": 6,
      "borderWidth": 1,
      "elevation": 5,
      "flexDirection": "row",
      "minHeight": 40,
      "paddingHorizontal": 12,
      "paddingVertical": 8,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 10,
      "transform": [
        {
          "translateY": 667,
        },
      ],
    }
  }
  testID="toast-container"
>
  <View
    accessible={false}
    style={
      [
        {
          "flex": 1,
          "shadowRadius": 30,
        },
      ]
    }
  >
    <Text
      accessibilityLabel="toast-text"
      accessible={true}
      style={
        [
          {
            "color": "#273142",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 16,
            "textAlign": "left",
          },
        ]
      }
      testID="toast-text"
    >
      lorem ipsum dolor sit amet.
    </Text>
  </View>
  <View
    accessibilityLabel="toast-action"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="toast-action"
  >
    <Text
      accessibilityLabel="toast-action-text"
      accessible={true}
      style={
        [
          {
            "color": "#273142",
            "fontFamily": "Rubik-Medium",
            "fontSize": 14,
            "fontWeight": "500",
            "lineHeight": 16,
            "padding": 8,
          },
        ]
      }
      testID="toast-action-text"
    >
      Tamam
    </Text>
  </View>
</View>
`;
